### ISSUE TITLE
How to run this code ?

### ISSUE THREAD

**Issue by basant01** [guilhermefloriani/signature-recognition#3]
Hi Can you please help me which file to run and where is the final output of the code ?

**Comment by gnbaron** [comment]
As the `README.md` file says you can just run one of the python scripts (i.e `python sigrecogtf.py`) and you will get just one single output, which is the score of the network on the specified author samples. Currently, the author is defined within the source file.

**Comment by vfbsilva** [comment]
@guilhermefloriani I'm on the same boat as @basant01 I've found the parameter you refer to. But it seems something is hardcoded or maybe you missed some file on the commit?

**Comment by gnbaron** [comment]
Well, as I said, the parameter that specifies the author is in the source file.

https://github.com/guilhermefloriani/signature-recognition/blob/5fe360116a1e58792464a0b47191cd7afe4eed35/sigrecogtf.py#L12

https://github.com/guilhermefloriani/signature-recognition/blob/5fe360116a1e58792464a0b47191cd7afe4eed35/sigrecog.py#L13

The training and test subsets will be selected using this author id.

**Comment by vfbsilva** [comment]
Yes I've noticed that but it seems form the tests I did that the evaluation is only considering the suffix of the files and not the NN. I've tried to train and use a set of signatures from Ayrton Senna I could find from autographs with no luck.

**Comment by gnbaron** [comment]
It does use the file prefix to train the NN and to assert the results as well.

**Comment by vfbsilva** [comment]
Do i need the prefix on my training data?

**Comment by gnbaron** [comment]
Yes, it should follow the same pattern. Otherwise, there's no way to know whether an example is genuine or forged.

**Comment by vfbsilva** [comment]
So I cannot classify a  sample having as input only valid ones?

Em sex, 9 de ago de 2019 17:44, guilhermefloriani <notifications@github.com>
escreveu:

**Comment by gnbaron** [comment]
Actually, it does not make much sense I think. How could the NN learn to identify a forged signature while you only train it w/ genuine samples?

It might be a way to do that but that's not the approach of this experiment.

What do you think @anisiomarxjr ?

**Comment by vfbsilva** [comment]
@guilhermefloriani Oki I was replying on mobile and that sucked, sorry. 

- What I did expect: Based on a set of valid signatures to be able to identify a forged one. 

- How: based on a given difference threshold. So I have a set of valid signatures and will consider invalid the ones that differ too much.

So the NN would say: xyz% sure. I still don't get how this code works at the moment, as I did not read it properly. I'm just space exploring and was looking for implementations. Yours has the biggest number of stars :)

**Comment by anisiomqs** [comment]
@vfbsilva @guilhermefloriani given the popularity of this repository I've been thinking for a while in approaching this problem differently. When we came up with this solution, we were not aware of the one-shot learning techniques.
Now that we have a better understanding of these techniques, it looks promising. 
In summary, we'll generate an embedding of the anchor signature, and use a norm to evaluate if new samples are real or forged. That's roughly how SOTA facial recognition algorithms work, and maybe if we take advantage of a convolution neural network trained for handwritten characters we could leverage transfer learning and reduce training time.
Let me know what you think of these ideas. This work would be published in a different repository.

**Comment by vfbsilva** [comment]
Perfect!

**Comment by basant01** [comment]
@vfbsilva  I think we should develop something like this because I had tried lots of signature verification repository but none of us performing well?

**Comment by gnbaron** [comment]
Cool! Sounds great @anisiomarxjr, let me know if I can help with something.

I’m sorry @vfbsilva, but we made this experiment a couple of years ago and this approach worked well in that controlled scenario. Hopefully the new approach mentioned by @anisiomarxjr works for you.

**Comment by vfbsilva** [comment]
@basant01 yes. I've found many repositories all of the seem to present some issue or having missing parts. Did you had any luck finding a proper one?
